eng/scripts/Validate-TypeSpec.ps1

```diff
@@ -25,8 +27,12 @@ if ($typespecFolders) {
     if ($LASTEXITCODE) {
       $exitCode = 1
     }
-    git restore .
-    git clean -df
+    if ($GitClean) {
```

Comment:
Should we have a warning message for devs that didn't turn on git clean?

Reply:
or a comment right above the parameter explaining it might do the job?

Reply:
I don't think it's necessary.  I think devs will almost always want `GitClean=false`, and CI will almost always want `GitClean=true`.